core/src/main/java/org/hibernate/ogm/util/impl/AssociationPersister.java

```diff
@@ -139,25 +137,56 @@ public AssociationKey getAssociationKey() {
 	}
 
 	/*
-	 * Load a collection and create it if it is not found
+	 * Load an association and create it if it is not found
 	 */
 	public Association getAssociation() {
 		if ( association == null ) {
 			AssociationKey key = getAssociationKey();
-			association = gridDialect.getAssociation( key, getAssociationContext() );
-			if (association == null) {
+
+			if ( hostingEntity != null ) {
+				OgmEntityEntryState entryState = OgmEntityEntryState.getStateFor( session, hostingEntity );
+				if ( entryState.hasAssociation( associationKeyMetadata.getCollectionRole() ) ) {
+					association = entryState.getAssociation( associationKeyMetadata.getCollectionRole() );
+				}
+				else {
+					association = gridDialect.getAssociation( key, getAssociationContext() );
+				}
+			}
+			else {
+				association = gridDialect.getAssociation( key, getAssociationContext() );
+			}
+
+			if ( association == null ) {
 				association = gridDialect.createAssociation( key, getAssociationContext() );
 			}
+			if ( hostingEntity != null ) {
```

Comment:
I also want to save it in the case at line 155. But you're right, it might not be a good idea to store it in every case just to save one call.